src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosFhirDataStore.cs

```diff
@@ -364,6 +375,37 @@ internal async Task<(IReadOnlyList<T> results, string continuationToken)> Execut
             return (results, page.ContinuationToken);
         }
 
+        private void UpdateSortIndex(FhirCosmosResourceWrapper cosmosWrapper)
+        {
+            Dictionary<string, SearchParameterInfo> searchParameters = _supportedSearchParameters.Value.GetSearchParameters(cosmosWrapper.ResourceTypeName)
+                .Where(x => x.SortStatus != SortParameterStatus.Disabled)
+                .ToDictionary(x => x.Code);
+
+            if (searchParameters.Any())
+            {
+                foreach (KeyValuePair<string, SortValue> item in cosmosWrapper.SortValues.ToArray())
+                {
+                    if (!searchParameters.ContainsKey(item.Key))
+                    {
+                        cosmosWrapper.SortValues.Remove(item.Key);
```

Comment:
Seems like a shame that we are finding min/max for search parameters where sort is not even enabled. Can we prune this uptream instead of here?